routes/user.go

```diff
@@ -3775,3 +3776,134 @@ func (fes *APIServer) GetProfileEntryResponseForPublicKeyBytes(publicKeyBytes []
 	}
 	return profileEntryResponse
 }
+
+type GetHoldersForPublicKeyWithLockedBalancesRequest struct {
+	// Either PublicKeyBase58Check or Username can be set by the client to specify
+	// which user we're obtaining posts for
+	// If both are specified, PublicKeyBase58Check will supercede
+	PublicKeyBase58Check string `safeForLogging:"true"`
+	Username             string `safeForLogging:"true"`
+}
+
+type GetHoldersForPublicKeyWithLockedBalancesResponse struct {
+	Holders map[string]*ExtendedBalanceEntryResponse
+}
+
+func (fes *APIServer) GetHoldersForPublicKeyWithLockedBalances(ww http.ResponseWriter, req *http.Request) {
+	decoder := json.NewDecoder(io.LimitReader(req.Body, MaxRequestBodySizeBytes))
+	requestData := GetHoldersForPublicKeyWithLockedBalancesRequest{}
+	if err := decoder.Decode(&requestData); err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf(
+			"GetHoldersForPublicKeyWithLockedBalances: Problem parsing request body: %v", err))
+		return
+	}
+
+	// Get a view
+	utxoView, err := fes.backendServer.GetMempool().GetAugmentedUniversalView()
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("GetHoldersForPublicKeyWithLockedBalances: Error getting utxoView: %v", err))
+		return
+	}
+
+	// Decode the public key for which we are fetching hodlers / hodlings.  If public key is not provided, use username
+	var creatorPublicKeyBytes []byte
+	if requestData.PublicKeyBase58Check != "" {
+		creatorPublicKeyBytes, _, err = lib.Base58CheckDecode(requestData.PublicKeyBase58Check)
+		if err != nil {
+			_AddBadRequestError(ww, fmt.Sprintf("GetHoldersForPublicKeyWithLockedBalances: Problem decoding user public key: %v", err))
+			return
+		}
+	} else {
+		username := requestData.Username
+		profileEntry := utxoView.GetProfileEntryForUsername([]byte(username))
+
+		// Return an error if we failed to find a profile entry
+		if profileEntry == nil {
+			_AddNotFoundError(ww, fmt.Sprintf("GetHoldersForPublicKeyWithLockedBalances: could not find profile for username: %v", username))
+			return
+		}
+		creatorPublicKeyBytes = profileEntry.PublicKey
+	}
+
+	unlockedBalanceEntrys, _, lockedBalanceEntryMap, _, err := utxoView.GetHolders(
```

Comment:
If this is the only place where we need both the unlocked and locked balance entries, I strongly recommend we break out getting the locked balance entries in core into its own function.